test/e2e/node/node_problem_detector.go

```diff
@@ -118,13 +126,20 @@ var _ = SIGDescribe("NodeProblemDetector", func() {
 		numIterations := 60
 		for i := 1; i <= numIterations; i++ {
 			for _, host := range hosts {
-				rss, workingSet := getMemoryStat(f, host)
-				rssStats[host] = append(rssStats[host], rss)
-				workingSetStats[host] = append(workingSetStats[host], workingSet)
-				if i == numIterations {
-					cpuUsage, uptime := getCpuStat(f, host)
-					cpuUsageStats[host] = append(cpuUsageStats[host], cpuUsage)
-					uptimeStats[host] = append(uptimeStats[host], uptime)
+				if isStandaloneMode[host] {
+					rss, workingSet := getMemoryStat(f, host)
+					rssStats[host] = append(rssStats[host], rss)
+					workingSetStats[host] = append(workingSetStats[host], workingSet)
+					if i == numIterations {
```

Comment:
it looks like this is a case of just keeping what was here before so I won't block

but FWIW it's not immediately clear to me why we wait until here to gather CPU stats in standalone mode vs. gathering them every iteration in pod mode

Reply:
When in standalone mode, NPD is running as systemd service instead of container. So we calculate the CPU usage from cgroup cpuacct differences. When NPD is a container, we just get the cpu usage from summary API.

I added some comment below.